hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/db/CodecRegistry.java

```diff
@@ -105,15 +109,17 @@ private <T> Codec getCodec(T object) throws IOException {
    */
   private <T> Codec getCodec(Class<T> format) throws IOException {
     Codec<T> codec;
-    if (valueCodecs.containsKey(format)) {
-      codec = (Codec<T>) valueCodecs.get(format);
-    } else if (valueCodecs.containsKey(format.getSuperclass())) {
-      codec = (Codec<T>) valueCodecs.get(format.getSuperclass());
-    } else {
-      throw new IllegalStateException(
-          "Codec is not registered for type: " + format);
```

Comment:
This adds codec support for interfaces implemented and not related to the Jira. Can you add a comment in the PR or Jira describing the issue?

Reply:
Also, there seems to be an implicit order for which codec will be used. I think this works but looks clunky.

Reply:
> This adds codec support for interfaces implemented and not related to the Jira.

It is related.  Generated c;ass for specific proto message extends `GeneratedMessage`, which implements `Message`.  
So generated class doesn't directly implement `Message`.  Thus we need to lookup by interface implemented by parent/ancestor.